draft-ietf-jsonpath-base.md

```diff
@@ -839,65 +830,76 @@ Queries:
+element-index   = int                             ; decimal integer
+
+int             = ["-"] ( "0" / (DIGIT1 *DIGIT) ) ; -  optional
+DIGIT1          = %x31-39                         ; 1-9 non-zero digit
 ~~~~
 
-#### Semantics
-{: unnumbered}
+Applying the numerical `element-index` is required to select the corresponding
+element. JSONPath allows it to be negative (see {{index-semantics}}).
 
-An `index-wild-selector`
-selects the nodes of all member values of an object as well as of all elements of an
-array.
+Notes:
+1. An `element-index` is an integer (in base 10, as in JSON numbers).
+2. As in JSON numbers, the syntax does not allow octal-like integers with leading zeros such as `01` or `-01`.
 
-Applying the `index-wild-selector` to a primitive JSON value (that is,
-a number, a string, `true`, `false`, or `null`) selects no node.
+##### Semantics {#index-semantics}
+{: unnumbered}
 
-The `index-wild-selector` behaves identically to the `dot-wild-selector`.
+The `element-index` applied to an array selects an array element using a zero-based index.
+For example, selector `0` selects the first and selector `4` the fifth element of a sufficiently long array.
+Nothing is selected, and it is not an error, if the index lies outside the range of the array. Nothing is selected from a value that is not an array.
 
-#### Examples
+A negative `element-index` counts from the array end.
+For example, selector `-1` selects the last and selector `-2` selects the penultimate element of an array with at least two elements.
+As with non-negative indexes, it is not an error if such an element does
+not exist; this simply means that no element is selected.
+
+##### Examples
 {: unnumbered}
 
+<!-- EDITING NOTE: There are non-breaking spaces here between j and j -->
+<!-- i.e., j j and not j j -->
+
 JSON:
 
     {
-      "o": {"j": 1, "k": 2},
-      "a": [5, 3]
+      "o": {"j j": {"k.k": 3}},
+      "a": ["a","b"],
+      "'": {"@": 2}
     }
 
 Queries:
 
 | Query | Result | Result Paths | Comment |
 | :---: | ------ | :----------: | ------- |
-| `$.o[*]` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
-| `$.o[*]` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
-| `$.a[*]` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Index wildcard selector examples"}
+| `$.a[1]`   | `"b"` | `$['a'][1]`      | Member of array      |
+| `$.a[-2]`   | `"a"` | `$['a'][0]`      | Member of array, from the end      |
+{: title="Index //PICKER// examples"}
 
-### Array Slice Selector {#slice}
+#### Array Slice //PICKER// {#slice}
 
-#### Syntax
+##### Syntax
 {: unnumbered}
 
-The array slice selector has the form `[<start>:<end>:<step>]`.
-It selects elements starting at index `<start>`, ending at — but
+The array slice //PICKER// has the form `<start>:<end>:<step>`.
+It selects elements from arrays starting at index `<start>`, ending at — but
```

Comment:
//PICKER// descriptions need to avoid saying that a //PICKER// can _select_ a value, as this confuses //PICKERS// and selectors.

Reply:
Seriously, are you going to type this out for every instance you see this?